src/constants/tokens.ts

```diff
@@ -356,12 +465,10 @@ export class ExtendedEther extends Ether {
 
 const cachedNativeCurrency: { [chainId: number]: NativeCurrency } = {}
 export function nativeOnChain(chainId: number): NativeCurrency {
-  return (
-    cachedNativeCurrency[chainId] ??
-    (cachedNativeCurrency[chainId] = isMatic(chainId)
-      ? new MaticNativeCurrency(chainId)
-      : ExtendedEther.onChain(chainId))
-  )
+  if (cachedNativeCurrency[chainId]) return cachedNativeCurrency[chainId]
+  if (isMatic(chainId)) return (cachedNativeCurrency[chainId] = new MaticNativeCurrency(chainId))
```

Comment:
can you simplify this with an if-else?

    let nativeCurrency: NativeCurrency
    if (isMatic(chainId) {
      nativeCurrency = new MaticNativeCurrency(chainId)
    } else if (isCelo(chainId) {
      ...
    } else {
      nativeCurrency = ExtendedEther.onChain(chainId)
    }
    return (cachedNativeCurrency[chainId] = nativeCurrency)